app/actions/handlers.py

```diff
@@ -0,0 +1,138 @@
+            with attempt:
+                auth = get_auth_config(integration)
+                device_list = await client.get_devices(integration, auth)
+                logger.info(f"Extracted {len(device_list)} devices from Lotek for inbound: {integration.id}")
+                observations_extracted = 0
+                for device in device_list:
+                    cdip_positions = []
+                    lower_date = max(present_time - timedelta(days=7), state.last_run)
+                    while lower_date < present_time:
+                        upper_date = min(present_time, lower_date + timedelta(days=7))
+                        positions = await client.get_positions(device.nDeviceID, auth, integration, lower_date, upper_date, True)
+                        logger.info(f"Extracted {len(positions)} obs from Lotek for device: {device.nDeviceID} between {lower_date} and {upper_date}.")
+                        for position in positions:
+                            cdip_pos = transform(position, integration)
+                            if cdip_pos:
+                                cdip_positions.append(cdip_pos)
+                        logger.debug(f"Extracted {len(cdip_positions)} of {len(positions)} points between {lower_date} and {upper_date}.")
+                        lower_date = upper_date
+
+                    if cdip_positions:
+                        logger.info(f"Observations pulled successfully for integration ID: {integration.id}.")
+
+                        def generate_batches(iterable, n=action_config.observations_per_request):
+                            for i in range(0, len(iterable), n):
+                                yield iterable[i: i + n]
+
+                        for i, batch in enumerate(generate_batches(cdip_positions)):
+                            async for attempt in stamina.retry_context(on=httpx.HTTPError, attempts=3, wait_initial=timedelta(seconds=10), wait_max=timedelta(seconds=10)):
+                                with attempt:
+                                    try:
+                                        logger.info(f'Sending observations batch #{i}: {len(batch)} observations. Device: {device.nDeviceID}')
+                                        await gundi_tools.send_observations_to_gundi(observations=batch, integration_id=integration.id)
+                                    except httpx.HTTPError as e:
+                                        msg = f'Sensors API returned error for integration_id: {integration.id}. Exception: {e}'
+                                        logger.exception(msg, extra={
+                                            'needs_attention': True,
+                                            'integration_id': integration.id,
+                                            'action_id': "pull_observations"
+                                        })
+                                        raise e
```

Comment:
I think we can remove all the retry logic here to simplify the code since `gundi_tools.send_observations_to_gundi` is already [handling retries with stamina](https://github.com/PADAS/gundi-integration-lotek/blob/4b0188f71eb09404070075c6e378da574a979dda/app/services/gundi.py#L72).